apps/core/factories.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+import factory
+import wagtail_factories
+from django.conf import settings
+from django.utils import timezone
+from wagtail.core.models import Locale, Page, Site
+
+from apps.core.models import ContentPage, HomePage
+
+
+class LocaleFactory(factory.django.DjangoModelFactory):
+    language_code = settings.LANGUAGE_CODE
+
+    class Meta:
+        model = Locale
+        django_get_or_create = ("language_code",)
+
+
+class HomePageFactory(wagtail_factories.PageFactory):
+    class Meta:
+        model = HomePage
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
+        home = HomePage.objects.first()
+        if not home:
+            root = Page.get_first_root_node()
+            home = HomePage(title="Home", slug="home-x")
+            root.add_child(instance=home)
+            site = Site.objects.first()
+            old_home = site.root_page
+            site.root_page = home
+            site.save()
+            old_home.delete()
+        return home
+
+
+class ContentPageFactory(wagtail_factories.PageFactory):
+    title = factory.Sequence(lambda n: "Page {}".format(n))
+    locale = factory.SubFactory(LocaleFactory)
+    body = json.dumps(
+        [
+            {"type": "text", "value": "some test content"},
+            {"type": "text", "value": "some more test content"},
+        ],
+    )
+
+    class Meta:
+        model = ContentPage
+
+    @factory.lazy_attribute
+    def first_published_at(self):
+        return timezone.now()
+
+    @factory.lazy_attribute
+    def parent(self):
+        return HomePageFactory()
```

Comment:
Because `first_published_at = timezone.now()` is evaluated once and gives all objects the same time:
```
In [10]: import factory
    ...:
    ...: class UserFactory(factory.Factory):
    ...:     last_login = timezone.now()
    ...:     class Meta:
    ...:         model = get_user_model()
    ...:

In [13]: user = UserFactory()

In [14]: user2 = UserFactory()

In [16]: user.last_login == user2.last_login
Out[16]: True
```
But we probably want a new datetime every time we call the factory. It is just more realistic.

For the homepage, we'd like to call the Factory every time too, just to make sure we have a home page.

Here are a couple of ways to do stuff lazy:
- https://factoryboy.readthedocs.io/en/stable/reference.html#lazyattribute
- https://factoryboy.readthedocs.io/en/stable/reference.html#lazyfunction
- https://factoryboy.readthedocs.io/en/stable/reference.html#lazyattributesequence

Reply:
Oh okay, got it. Thanks :+1: